tests/handlers/test_oidc.py

```diff
@@ -855,69 +854,60 @@ def _generate_oidc_session_token(
         )
 
 
+@skip_unless(HAS_OIDC, "requires OIDC")
 class UsernamePickerTestCase(HomeserverTestCase):
-    if not HAS_OIDC:
-        skip = "requires OIDC"
+    """Tests for the username picker flow of SSO login"""
 
     servlets = [login.register_servlets]
 
     def default_config(self):
         config = super().default_config()
         config["public_baseurl"] = BASE_URL
-        oidc_config = {
-            "enabled": True,
-            "client_id": CLIENT_ID,
-            "client_secret": CLIENT_SECRET,
-            "issuer": ISSUER,
-            "scopes": SCOPES,
-            "user_mapping_provider": {
-                "config": {"display_name_template": "{{ user.displayname }}"}
-            },
-        }
 
-        # Update this config with what's in the default config so that
-        # override_config works as expected.
-        oidc_config.update(config.get("oidc_config", {}))
-        config["oidc_config"] = oidc_config
+        config["oidc_config"] = {}
+        config["oidc_config"].update(TEST_OIDC_CONFIG)
```

Comment:
Is this just to make a copy? Would it be clearer to do `config["oidc_config"] = TEST_OIDC_CONFIG.copy()`?

Reply:
I guess... I tend to forget about the existence of `dict.copy` :/